#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.9999906737293804 0.0024262529636310684 -0.0035729190889231786 4.673829014736041
  position -0.11354859256218564 4.775979733855019 0.030783717595237854
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 3
  floorAppearance NULL
}
E-puck {
  translation 0.14 0 1.1
  controller "LineFollower"
  groundSensorsSlot [
    Camera {
      translation -0.0435632 1.32332e-05 0.002
      rotation 0.2507530721122057 0.25055107205411403 -0.9350652689084463 -1.6370853071795866
      name "CAM"
      fieldOfView 1
      width 256
      height 256
    }
    DistanceSensor {
      translation -0.01 -0.004 -0.006
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_ML"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
    DistanceSensor {
      translation -0.01 0.004 -0.006
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_MR"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
    DistanceSensor {
      translation -0.01 -0.008 -0.006
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_L"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
    DistanceSensor {
      translation -0.01 0.008 -0.006
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_R"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
    DistanceSensor {
      translation -0.01 0.024 0.01
      rotation 1 0 0 1.5708
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_RS"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
    DistanceSensor {
      translation -0.01 -0.024 0.0099998
      rotation 1 0 0 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 0.4
            metalness 0
          }
          geometry Box {
            size 0.002 0.002 0.004
          }
        }
      ]
      name "ir_LS"
      lookupTable [
        0 4095 0.002
        0.005 2133.33 0.003
        0.01 1465.73 0.007
        0.015 601.46 0.0406
        0.02 383.84 0.01472
        0.03 234.93 0.0241
        0.04 158.03 0.0287
        0.05 120 0.04225
        0.06 104.09 0.03065
        0.07 67.19 0.04897
      ]
      type "infra-red"
      aperture 0.1
    }
  ]
}
DEF META Solid {
  translation 0.07 0.001 -0.87
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Plane {
        size 0.02 0.02
      }
    }
  ]
  name "meta"
}
DEF TRACK2 Solid {
  rotation 1 0 0 -1.5707953071795862
  scale 0.01 0.01 0.01
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
            name "color_2829873"
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                18 49.784 0, 46.349 21.565 0, 17.086 48 0, 18.065 49.849 0, -13 -57.586 0.01, -14.784 -58.5 0.01, 13 -86.414 0.01, -14.849 -58.565 0.01, 46.5 21.5 0.01, 44.5 -18 0.01, 46.5 -20 0.01, 8 66.5 0, -10 66.5 0, 6 68.5 0, -15 -18 0, -13 -20 0, -15 -58.5 0, 14.849 -85.435 0.01, 14.784 -85.5 0.01, 15 -96 0.01, 8 66.5 0.01, 8 86.5 0.01, -10 66.5 0.01, -10 50 0.01, -10 50 0, -14.784 -58.5 0, 14.849 -85.435 0, 14.784 -85.5 0, -13 -57.586 0, 13 -86.414 0, 13 -96 0.01, -15 -18 0.01, -15 -58.5 0.01, 18.065 49.849 0.01, 18 49.784 0.01, -12 48 0, 44.5 20.586 0, 44.5 -18 0, 44.5 20.586 0.01, -14.849 -58.565 0, 46.349 21.565 0.01, 46.284 21.5 0.01, 13 -96 0, 46.284 21.5 0, 46.5 -20 0, 15 -85.5 0.01, 15 -85.5 0, 46.5 21.5 0, 15 -96 0, -13 -20 0.01, 18 50 0, 18 50 0.01, -12 68.5 0.01, -12 68.5 0, 17.086 48 0.01, 6 86.5 0.01, 6 86.5 0, 6 68.5 0.01, -12 48 0.01, 8 86.5 0
              ]
            }
            coordIndex [
              0, 1, 2, -1, 3, 1, 0, -1, 4, 5, 6, -1, 7, 6, 5, -1, 8, 9, 10, -1, 11, 12, 13, -1, 14, 15, 16, -1, 17, 4, 18, -1, 6, 18, 4, -1, 19, 18, 6, -1, 20, 11, 21, -1, 22, 23, 24, -1, 22, 24, 12, -1, 16, 15, 25, -1, 26, 27, 28, -1, 27, 29, 28, -1, 30, 19, 6, -1, 31, 14, 32, -1, 16, 32, 14, -1, 33, 3, 0, -1, 33, 0, 34, -1, 2, 35, 24, -1, 5, 16, 25, -1, 36, 37, 9, -1, 36, 9, 38, -1, 5, 25, 39, -1, 5, 39, 7, -1, 14, 37, 15, -1, 40, 33, 41, -1, 4, 17, 26, -1, 29, 42, 30, -1, 29, 30, 6, -1, 38, 9, 41, -1, 8, 41, 9, -1, 36, 2, 43, -1, 44, 15, 37, -1, 45, 46, 27, -1, 45, 27, 18, -1, 18, 27, 26, -1, 18, 26, 17, -1, 36, 43, 37, -1, 1, 43, 2, -1, 47, 37, 43, -1, 48, 27, 46, -1, 49, 4, 28, -1, 49, 28, 15, -1, 48, 29, 27, -1, 25, 28, 29, -1, 48, 42, 29, -1, 39, 25, 29, -1, 30, 42, 48, -1, 30, 48, 19, -1, 41, 43, 1, -1, 41, 1, 40, -1, 4, 49, 5, -1, 31, 32, 49, -1, 43, 41, 47, -1, 8, 47, 41, -1, 34, 0, 50, -1, 34, 50, 51, -1, 28, 25, 15, -1, 33, 40, 1, -1, 31, 9, 14, -1, 14, 9, 37, -1, 1, 3, 33, -1, 4, 26, 28, -1, 15, 44, 10, -1, 24, 51, 50, -1, 47, 44, 37, -1, 51, 23, 34, -1, 23, 22, 52, -1, 10, 44, 47, -1, 10, 47, 8, -1, 2, 24, 0, -1, 50, 0, 24, -1, 13, 12, 53, -1, 24, 53, 12, -1, 32, 16, 5, -1, 2, 36, 38, -1, 2, 38, 54, -1, 52, 13, 53, -1, 46, 45, 48, -1, 19, 48, 45, -1, 39, 29, 6, -1, 39, 6, 7, -1, 24, 23, 51, -1, 55, 56, 13, -1, 55, 13, 57, -1, 2, 54, 35, -1, 58, 35, 54, -1, 54, 38, 34, -1, 33, 34, 38, -1, 38, 41, 33, -1, 59, 21, 11, -1, 34, 23, 54, -1, 58, 54, 23, -1, 52, 58, 23, -1, 32, 5, 49, -1, 21, 55, 57, -1, 56, 59, 13, -1, 11, 13, 59, -1, 53, 24, 35, -1, 52, 53, 35, -1, 52, 35, 58, -1, 21, 59, 56, -1, 21, 56, 55, -1, 19, 45, 18, -1, 13, 52, 57, -1, 10, 49, 15, -1, 10, 9, 49, -1, 31, 49, 9, -1, 11, 20, 22, -1, 11, 22, 12, -1, 20, 57, 22, -1, 21, 57, 20, -1, 52, 22, 57, -1
            ]
          }
        }
      ]
      name "group_0_2829873"
    }
  ]
  name "track2.obj(1)"
}
